code/game/atoms/atoms_movable.dm

```diff
@@ -533,4 +533,10 @@
 	qdel(em_render)
 	em_render = null
 
+/**
+ * Checks if we can avoid things like landmine, lava, etc, whether beneficial or harmful.
+ */
+/atom/movable/proc/is_avoiding_ground()
+    return (movement_type != GROUND) || throwing
+
```

Comment:
please shove this below the comment below it
the comment below it refers to the emissives stuff

Reply:
Shifted down!